src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/RazorCohostDynamicRegistrationService.cs

```diff
@@ -39,20 +39,18 @@ public async Task RegisterAsync(string clientCapabilitiesString, RazorCohostRequ
             return;
         }
 
-        // TODO: Should we delay everything below this line until a Razor file is opened?
-
         var clientCapabilities = JsonSerializer.Deserialize<VSInternalClientCapabilities>(clientCapabilitiesString) ?? new();
 
         _lazyRazorCohostClientCapabilitiesService.Value.SetCapabilities(clientCapabilities);
 
+        // We assume most registration providers will just return one, so whilst this isn't completely accurate, it's a
+        // reasonable starting point
         _lazyRegistrationProviders.TryGetCount(out var providerCount);
```

Comment:
Consider changing the type of `_lazyRegistrationProviders` to `ImmutableArray<Lazy<IDynamicRegistrationProvider>>` and changing its field initializer to `lazyRegistrationProviders.ToImmutableArray()`. Then, you don't have to mess around with `TryGetCount` here and can avoid an enumerator allocation in the foreach below.